### DISCUSSION TITLE
Migrate *.tt file instructions to the new CLI scaffolding tool

### DISCUSSION THREAD

**Discussion by OutgoingOpossum** [linq2db/linq2db#3951] [Q&A] [answered]
What is the recommended way to migrate the following *.tt file instructions to the new CLI scaffolding tool?

### Association Property Names
```csharp
GetFK("public.building", "public.building_property_id_fk").BackReference.MemberName = "Buildings";
```

### 1:1 Associations
```csharp
GetFK("public.building_attribute", "public.building_attribute_building_fk").AssociationType = AssociationType.OneToOne;
```

### Renaming of Reserved Keywords
```csharp
GetColumn("public.building_attribute", "class").MemberName = "BuildingAttributeClass";
```

### Explicit Type Override
```csharp
GetColumn("public.commune", "geog").Type = "GeoJSONObject";
```

I could not find any corresponding CLI option. The only solution I have found so far is implementing a custom `ScaffoldInterceptors` and using specific static instructions in code. This doesn't feel right and screams for a more declarative way like a JSON config.

**Comment by MaceWindu** [comment, marked as answer]
1. `AssociationType` is removed from linq2db, because it had no functional role
2. To rename member or change specific member type use data model interceptors https://linq2db.github.io/articles/CLI.html#data-model-interceptors
3. To change database type mapping use type mapping interceptor https://linq2db.github.io/articles/CLI.html#type-mapping-interceptor
